src/ucp/api/ucp.h

```diff
@@ -1133,6 +1135,11 @@ ucs_status_t ucp_worker_get_efd(ucp_worker_h worker, int *fd);
  * This routine waits (blocking) until an event has happened, as part of the
  * wake-up mechanism.
  *
+ * This function is guaranteed to return only if new communication events occur
+ * on the @a worker. Therefore one must drain all existing events before waiting
+ * on the file descriptor. This can be achieved by by calling
+ * @ref ucp_worker_progress repeatedly until it returns 0.
+ *
  * There are two alternative ways to use the wakeup mechanism: the first is the
  * file descriptor obtained per worker using @ref ucp_worker_get_efd and the
  * second is waiting on the next event internally (this function).
```

Comment:
This text wasn't modified as part of the PR, but I suggest the following re-wording for clarity:

There are two alternative ways to use the wakeup mechanism. The first is by polling on a per-worker file descriptor obtained from @ref ucp_worker_get_efd. The second is by using this function to perform an internal wait for the next event associated with the specified worker.